gobblin-core/src/main/java/gobblin/writer/SimpleDataWriter.java

```diff
@@ -0,0 +1,175 @@
+      this.recordDelimiter = null;
+    } else {
+      this.recordDelimiter = delim.getBytes()[0];
+    }
+
+    this.outputStream = this.fs.create(this.stagingFile, true);
+    this.prependSize = properties.getPropAsBoolean(ConfigurationKeys.SIMPLE_WRITER_PREPEND_SIZE, true);
+    this.recordsWritten = 0;
+    this.bytesWritten = 0;
+    this.closed = false;
+  }
+  /**
+   * Write a source record to the staging file
+   *
+   * @param record data record to write
+   * @throws java.io.IOException if there is anything wrong writing the record
+   */
+  @Override
+  public void write(byte[] record) throws IOException {
+    byte[] toWrite = record;
+    Preconditions.checkNotNull(record);
+    if (recordDelimiter != null) {
+      toWrite = Arrays.copyOf(record, record.length + 1);
+      toWrite[toWrite.length - 1] = recordDelimiter;
+    }
+    if (prependSize) {
+      Long recordSize = new Long(toWrite.length);
+      ByteBuffer buf = ByteBuffer.allocate(Long.SIZE / 8);
+      buf.putLong(recordSize);
+      toWrite = ArrayUtils.addAll(buf.array(), toWrite);
+    }
+    this.outputStream.write(toWrite);
+    bytesWritten += (toWrite.length);
+    recordsWritten++;
+  }
+
+  /**
+   * Commit the data written to the final output file.
+   *
+   * @throws java.io.IOException if there is anything wrong committing the output
+   */
+  @Override
+  public void commit() throws IOException {
+    this.close();
+    if (!this.fs.exists(this.stagingFile)) {
+      throw new IOException(String.format("File %s does not exist", this.stagingFile));
+    }
+
+    LOG.info(String.format("Moving data from %s to %s", this.stagingFile, this.outputFile));
+    // For the same reason as deleting the staging file if it already exists, deleting
+    // the output file if it already exists prevents task retry from being blocked.
+    if (this.fs.exists(this.outputFile)) {
+      LOG.warn(String.format("Task output file %s already exists", this.outputFile));
+      this.deletePath(this.outputFile, false);
+    }
+    this.fs.rename(this.stagingFile, this.outputFile);
+  }
+
+  /**
+   * Cleanup context/resources.
+   *
+   * @throws java.io.IOException if there is anything wrong doing cleanup.
+   */
+  @Override
+  public void cleanup() throws IOException {
+    if (this.fs.exists(this.stagingFile)) {
+      this.deletePath(this.stagingFile, false);
+    }
+  }
+
+  /**
+   * Get the number of records written.
+   *
+   * @return number of records written
+   */
+  @Override
+  public long recordsWritten() {
+    return this.recordsWritten;
+  }
+
+  /**
+   * Get the number of bytes written.
+   * <p/>
+   * <p>
+   * This method should ONLY be called after {@link gobblin.writer.DataWriter#commit()}
+   * is called.
+   * </p>
+   *
+   * @return number of bytes written
+   */
+  @Override
+  public long bytesWritten() throws IOException {
+    if (this.closed) {
+      return this.bytesWritten;
+    } else {
+      return 0;
```

Comment:
The javadoc is copied from the interface, so not sure what the intent here was. The AvroHdfsDataWriter provided out of box does not throw an exception, but returns 0 so that behavior was copied here.